hibernate-core/src/main/java/org/hibernate/internal/AbstractSharedSessionContract.java

```diff
@@ -152,8 +153,12 @@
 
 	private CriteriaCompiler criteriaCompiler;
 
-	public AbstractSharedSessionContract(SessionFactoryImpl factory, SessionCreationOptions options) {
+	public AbstractSharedSessionContract(
+			SessionFactoryImpl factory,
+			SessionCreationOptions options,
+			FastSessionServices fastSessionServices) {
```

Comment:
nitpicking: probably we can avoid changing the constructor, we can get the `FastSessionServices` from the `factory`

Reply:
+1

thanks, cleaned it up and pushed a new version.